sumatra/core.py

```diff
@@ -105,8 +125,67 @@ def register(self, component):
                     name = component.name
                 else:
                     name = component.__name__
-                self.components[base_class][name] = component
+                self._components[base_class][name] = component
                 return
         raise TypeError("%s is not a Sumatra component." % component)
 
-registry = Registry()
+
+def component_type(cls):
```

Comment:
Right. I implented that by issuing a warning instead of raising an error to not be to restrictive on this. I guess you can think of wild use cases where one wants to subclass a component type to create a new component type.